tests/system/src/test/java/test/com/sun/prism/impl/NGNodeDirtyFlagTest.java

```diff
@@ -0,0 +1,188 @@
+import javafx.scene.image.WritableImage;
+import javafx.scene.robot.Robot;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javafx.application.Platform;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Line;
+import javafx.stage.Stage;
+import test.util.Util;
+
+import static org.junit.Assert.assertEquals;
+
+public class NGNodeDirtyFlagTest {
+
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Singleton Application instance
+    static MyApp myApp;
+
+    private CountDownLatch latch = new CountDownLatch(1);
+
+    public static class MyApp extends Application {
+
+        private StackPane root;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            root = new StackPane();
+            primaryStage.setScene(new Scene(root, 500, 400));
+            primaryStage.show();
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, MyApp.class);
+        assertEquals(0, launchLatch.getCount());
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void testNGNodesNotDirty() throws InterruptedException {
+        ObjectProperty<Color> lineColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+        ObjectProperty<Color> circleColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+
+        StackPane root = myApp.root;
+
+        runAndWait(() -> {
+            var contents = new HBox();
+            contents.setSpacing(10);
+            contents.setPadding(new Insets(10));
+            contents.getChildren().add(contentElement("L", lineColor, circleColor));
+            contents.getChildren().add(contentElement("R", lineColor, circleColor));
+            root.getChildren().add(contents);
+
+            Pane sideArea = createSideArea();
+            StackPane.setAlignment(sideArea, Pos.CENTER_RIGHT);
+            root.getChildren().add(sideArea);
+        });
+
+        Thread.sleep(500);
```

Comment:
Could use `Util.waitForIdle(Scene)` instead of sleep. It would reduce the test time and should work. We can consider a sleep if `Util.waitForIdle(Scene)` does not work. Similarly please see if other sleep calls can be removed.